app/models/concerns/foreman_teamdynamix/host_extensions.rb

```diff
@@ -16,7 +17,14 @@ def td_api
     def create_teamdynamix_asset
       asset = td_api.create_asset(self)
       self.teamdynamix_asset_id = asset['ID']
+    rescue StandardError => e
+      errors.add(:base, e.message)
```

Comment:
Please follow this example: https://github.com/theforeman/puppetdb_foreman/blob/master/app/models/puppetdb_foreman/host_extensions.rb#L22

Use the language translation method: _()
Also return false. Some things in rails expect this on failure.